src/main/java/uk/gov/hmcts/reform/civil/service/citizen/HWFFeePaymentOutcomeService.java

```diff
@@ -3,58 +3,84 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
 import uk.gov.hmcts.reform.civil.enums.YesOrNo;
 import uk.gov.hmcts.reform.civil.model.CaseData;
 import uk.gov.hmcts.reform.civil.model.citizenui.HelpWithFees;
+import uk.gov.hmcts.reform.civil.service.citizenui.HelpWithFeesForTabService;
 import uk.gov.hmcts.reform.civil.utils.MonetaryConversions;
 
 import java.math.BigDecimal;
 import java.util.Objects;
 
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NO_REMISSION_HWF;
+
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class HWFFeePaymentOutcomeService {
 
+    private final HelpWithFeesForTabService helpWithFeesForTabService;
+
     public CaseData updateHwfReferenceNumber(CaseData caseData) {
+        CaseData.CaseDataBuilder<?, ?> updatedData = caseData.toBuilder();
 
         if (Objects.nonNull(caseData.getFeePaymentOutcomeDetails())
             && caseData.getFeePaymentOutcomeDetails().getHwfNumberAvailable() == YesOrNo.YES) {
             if (caseData.isHWFTypeClaimIssued()) {
+                var caseDataLip = caseData.getCaseDataLiP();
                 HelpWithFees helpWithFees = HelpWithFees.builder()
                     .helpWithFee(YesOrNo.YES)
                     .helpWithFeesReferenceNumber(caseData.getFeePaymentOutcomeDetails().getHwfNumberForFeePaymentOutcome())
                     .build();
-                caseData.getCaseDataLiP().setHelpWithFees(helpWithFees);
+                updatedData.caseDataLiP(caseDataLip.toBuilder().helpWithFees(helpWithFees).build());
+                helpWithFeesForTabService.setUpHelpWithFeeTab(updatedData);
             }
             if (caseData.isHWFTypeHearing()) {
-                caseData = caseData.toBuilder().hearingHelpFeesReferenceNumber(caseData.getFeePaymentOutcomeDetails().getHwfNumberForFeePaymentOutcome()).build();
+                updatedData.hearingHelpFeesReferenceNumber(caseData.getFeePaymentOutcomeDetails().getHwfNumberForFeePaymentOutcome()).build();
             }
-            clearHwfReferenceProperties(caseData);
+            clearHwfReferenceProperties(updatedData);
         }
-        return caseData;
+        return updatedData.build();
     }
 
-    private void clearHwfReferenceProperties(CaseData caseData) {
-        caseData.getFeePaymentOutcomeDetails().setHwfNumberAvailable(null);
-        caseData.getFeePaymentOutcomeDetails().setHwfNumberForFeePaymentOutcome(null);
+    private void clearHwfReferenceProperties(CaseData.CaseDataBuilder caseDataBuilder) {
+        CaseData caseData = caseDataBuilder.build();
+        caseDataBuilder.feePaymentOutcomeDetails(caseData.getFeePaymentOutcomeDetails().toBuilder().hwfNumberAvailable(null).build());
```

Comment:
No Need of 2 lines 49 & 50, can be combine. Currently only I property is being updated.

Reply:
Updated from review